File: src/Domain/Entities/TestResult.cs
using Domain.Enums;

namespace Domain.Entities;

public class TestResult : BaseEntity
{
    public TestResultStatus Status { get; set; }
    public string? Notes { get; set; }
    public DateTime ExecutedAt { get; set; }
    public Guid? ExecutedById { get; set; }

    public Guid TestRunId { get; set; }
    public TestRun TestRun { get; set; } = null!;

    public Guid TestCaseId { get; set; }
    public TestCase TestCase { get; set; } = null!;
}

File: apps/api/src/domain/test-result.ts
import { TestResultStatus } from "@testcraft/types";

export interface TestResult {
  id: string;
  testRunId: string;
  testCaseId: string;
  suiteId: string;
  testCaseName: string;
  status: TestResultStatus;
  notes: string | null;
  executedAt: Date;
  executedById: string | null;
  createdAt: Date;
  updatedAt: Date;
}
